<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>DebuggerWindow</class>
 <widget class="QMainWindow" name="DebuggerWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1210</width>
    <height>800</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>CPU Debugger</string>
  </property>
  <property name="dockNestingEnabled">
   <bool>true</bool>
  </property>
  <widget class="QWidget" name="centralwidget"/>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1210</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu_Debugger">
    <property name="title">
     <string>&amp;Debug</string>
    </property>
    <addaction name="actionPause"/>
    <addaction name="actionRunToCursor"/>
    <addaction name="separator"/>
    <addaction name="actionGoToPC"/>
    <addaction name="actionGoToAddress"/>
    <addaction name="actionDumpAddress"/>
    <addaction name="separator"/>    
    <addaction name="actionTrace"/>    
    <addaction name="separator"/>
    <addaction name="actionStepInto"/>
    <addaction name="actionStepOver"/>
    <addaction name="actionStepOut"/>
    <addaction name="separator"/>
    <addaction name="actionClose"/>

   </widget>
   <widget class="QMenu" name="menuBreakpoints">
    <property name="title">
     <string>Breakpoints</string>
    </property>
    <addaction name="actionAddBreakpoint"/>
    <addaction name="actionToggleBreakpoint"/>
    <addaction name="actionClearBreakpoints"/>
   </widget>
   <addaction name="menu_Debugger"/>
   <addaction name="menuBreakpoints"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionPause"/>
   <addaction name="actionRunToCursor"/>
   <addaction name="separator"/>
   <addaction name="actionGoToPC"/>
   <addaction name="actionGoToAddress"/>
   <addaction name="actionDumpAddress"/>
   <addaction name="separator"/>
   <addaction name="actionAddBreakpoint"/>
   <addaction name="actionToggleBreakpoint"/>
   <addaction name="actionClearBreakpoints"/>
   <addaction name="separator"/>
   <addaction name="actionStepInto"/>
   <addaction name="actionStepOver"/>
   <addaction name="actionStepOut"/>
   <addaction name="separator"/>   
   <addaction name="actionTrace"/>
  </widget>
  <widget class="QDockWidget" name="dockWidget_2">
   <property name="features">
    <set>QDockWidget::DockWidgetFloatable|QDockWidget::DockWidgetMovable</set>
   </property>
   <property name="windowTitle">
    <string>Disassembly</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>4</number>
   </attribute>
   <widget class="QTreeView" name="codeView">
    <property name="sizePolicy">
     <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
      <horstretch>0</horstretch>
      <verstretch>0</verstretch>
     </sizepolicy>
    </property>
    <attribute name="headerVisible">
     <bool>false</bool>
    </attribute>
   </widget>
  </widget>
  <widget class="QDockWidget" name="dockWidget_3">
   <property name="features">
    <set>QDockWidget::DockWidgetFloatable|QDockWidget::DockWidgetMovable</set>
   </property>
   <property name="windowTitle">
    <string>Registers</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>4</number>
   </attribute>
   <widget class="QTreeView" name="registerView">
    <property name="sizePolicy">
     <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
      <horstretch>0</horstretch>
      <verstretch>0</verstretch>
     </sizepolicy>
    </property>
    <property name="maximumSize">
     <size>
      <width>220</width>
      <height>16777215</height>
     </size>
    </property>
   </widget>
  </widget>
  <widget class="QDockWidget" name="dockWidget">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
     <horstretch>1</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="features">
    <set>QDockWidget::DockWidgetFloatable|QDockWidget::DockWidgetMovable</set>
   </property>
   <property name="windowTitle">
    <string>Memory</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>8</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_3">
    <layout class="QVBoxLayout" name="verticalLayout">
     <item>
      <widget class="MemoryViewWidget" name="memoryView" native="true">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
      </widget>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <item>
        <widget class="QRadioButton" name="memoryRegionRAM">
         <property name="text">
          <string>RAM</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QRadioButton" name="memoryRegionScratchpad">
         <property name="text">
          <string>Scratchpad</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QRadioButton" name="memoryRegionEXP1">
         <property name="text">
          <string>EXP1</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QRadioButton" name="memoryRegionBIOS">
         <property name="text">
          <string>BIOS</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <widget class="QLineEdit" name="memorySearchString"/>
         </item>
         <item>
          <widget class="QPushButton" name="memorySearch">
           <property name="text">
            <string>Search</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="dockWidget_5">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="maximumSize">
    <size>
     <width>200</width>
     <height>524287</height>
    </size>
   </property>
   <property name="features">
    <set>QDockWidget::DockWidgetFloatable|QDockWidget::DockWidgetMovable</set>
   </property>
   <property name="windowTitle">
    <string>Breakpoints</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>8</number>
   </attribute>
   <widget class="QTreeWidget" name="breakpointsWidget">
    <property name="selectionMode">
     <enum>QAbstractItemView::SingleSelection</enum>
    </property>
    <property name="selectionBehavior">
     <enum>QAbstractItemView::SelectRows</enum>
    </property>
    <property name="sortingEnabled">
     <bool>true</bool>
    </property>
    <attribute name="headerMinimumSectionSize">
     <number>20</number>
    </attribute>
    <column>
     <property name="text">
      <string>#</string>
     </property>
    </column>
    <column>
     <property name="text">
      <string>Address</string>
     </property>
    </column>
    <column>
     <property name="text">
      <string>Hit Count</string>
     </property>
    </column>
   </widget>
  </widget>
  <widget class="QDockWidget" name="dockWidget_4">
   <property name="features">
    <set>QDockWidget::DockWidgetFloatable|QDockWidget::DockWidgetMovable</set>
   </property>
   <property name="windowTitle">
    <string>Stack</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>8</number>
   </attribute>
   <widget class="QTreeView" name="stackView">
    <property name="maximumSize">
     <size>
      <width>220</width>
      <height>16777215</height>
     </size>
    </property>
   </widget>
  </widget>
  <action name="actionPause">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="resources/resources.qrc">
     <normaloff>:/icons/media-playback-pause.png</normaloff>:/icons/media-playback-pause.png</iconset>
   </property>
   <property name="text">
    <string>Pause/Continue</string>
   </property>
   <property name="toolTip">
    <string>&amp;Pause/Continue</string>
   </property>
   <property name="shortcut">
    <string>F5</string>
   </property>
  </action>
  <action name="actionStepInto">
   <property name="icon">
    <iconset resource="resources/resources.qrc">
     <normaloff>:/icons/debug-step-into-instruction.png</normaloff>:/icons/debug-step-into-instruction.png</iconset>
   </property>
   <property name="text">
    <string>Step Into</string>
   </property>
   <property name="toolTip">
    <string>&amp;Step Into</string>
   </property>
   <property name="shortcut">
    <string>F11</string>
   </property>
  </action>
  <action name="actionStepOver">
   <property name="icon">
    <iconset resource="resources/resources.qrc">
     <normaloff>:/icons/debug-step-over.png</normaloff>:/icons/debug-step-over.png</iconset>
   </property>
   <property name="text">
    <string>Step Over</string>
   </property>
   <property name="toolTip">
    <string>Step &amp;Over</string>
   </property>
   <property name="shortcut">
    <string>F10</string>
   </property>
  </action>
  <action name="actionToggleBreakpoint">
   <property name="icon">
    <iconset resource="resources/resources.qrc">
     <normaloff>:/icons/media-record@2x.png</normaloff>:/icons/media-record@2x.png</iconset>
   </property>
   <property name="text">
    <string>Toggle Breakpoint</string>
   </property>
   <property name="toolTip">
    <string>Toggle &amp;Breakpoint</string>
   </property>
   <property name="shortcut">
    <string>F9</string>
   </property>
  </action>
  <action name="actionClose">
   <property name="text">
    <string>&amp;Close</string>
   </property>
  </action>
  <action name="actionStepOut">
   <property name="icon">
    <iconset resource="resources/resources.qrc">
     <normaloff>:/icons/debug-step-out.png</normaloff>:/icons/debug-step-out.png</iconset>
   </property>
   <property name="text">
    <string>Step Out</string>
   </property>
   <property name="toolTip">
    <string>Step O&amp;ut</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+F11</string>
   </property>
  </action>
  <action name="actionRunToCursor">
   <property name="icon">
    <iconset resource="resources/resources.qrc">
     <normaloff>:/icons/debug-run-cursor.png</normaloff>:/icons/debug-run-cursor.png</iconset>
   </property>
   <property name="text">
    <string>Run To Cursor</string>
   </property>
   <property name="toolTip">
    <string>&amp;Run To Cursor</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+F10</string>
   </property>
  </action>
  <action name="actionClearBreakpoints">
   <property name="icon">
    <iconset resource="resources/resources.qrc">
     <normaloff>:/icons/edit-clear-16.png</normaloff>:/icons/edit-clear-16.png</iconset>
   </property>
   <property name="text">
    <string>Clear Breakpoints</string>
   </property>
   <property name="toolTip">
    <string>&amp;Clear Breakpoints</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Del</string>
   </property>
  </action>
  <action name="actionAddBreakpoint">
   <property name="icon">
    <iconset resource="resources/resources.qrc">
     <normaloff>:/icons/list-add.png</normaloff>:/icons/list-add.png</iconset>
   </property>
   <property name="text">
    <string>Add Breakpoint</string>
   </property>
   <property name="toolTip">
    <string>Add &amp;Breakpoint</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+F9</string>
   </property>
  </action>
  <action name="actionGoToPC">
   <property name="icon">
    <iconset resource="resources/resources.qrc">
     <normaloff>:/icons/document-open.png</normaloff>:/icons/document-open.png</iconset>
   </property>
   <property name="text">
    <string>Go To PC</string>
   </property>
   <property name="toolTip">
    <string>&amp;Go To PC</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+P</string>
   </property>
  </action>
  <action name="actionGoToAddress">
   <property name="icon">
    <iconset resource="resources/resources.qrc">
     <normaloff>:/icons/applications-system.png</normaloff>:/icons/applications-system.png</iconset>
   </property>
   <property name="text">
    <string>Go To Address</string>
   </property>
   <property name="toolTip">
    <string>Go To &amp;Address</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+G</string>
   </property>
  </action>
  <action name="actionDumpAddress">
   <property name="icon">
    <iconset resource="resources/resources.qrc">
     <normaloff>:/icons/antialias-icon.png</normaloff>:/icons/antialias-icon.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Dump Address</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+D</string>
   </property>
  </action>
  <action name="actionTrace">
   <property name="icon">
    <iconset resource="resources/resources.qrc">
     <normaloff>:/icons/debug-trace.png</normaloff>:/icons/debug-trace.png</iconset>
   </property>
   <property name="text">
    <string>Trace</string>
   </property>
   <property name="toolTip">
    <string>&amp;Trace</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+T</string>
   </property>
  </action>
  
  
 </widget>
 <customwidgets>
  <customwidget>
   <class>MemoryViewWidget</class>
   <extends>QWidget</extends>
   <header>duckstation-qt/memoryviewwidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="resources/resources.qrc"/>
 </resources>
 <connections/>
</ui>
